File: puzzles/python3/network_cabling.py
import math


if __name__ == "__main__":
    nb_buildings = int(input())
    cable_length = 0
    min_x = float("inf")
    max_x = 0
    y_coordinates = []

    for _ in range(nb_buildings):
        x, y = map(int, input().split())
        y_coordinates.append(y)
        if x < min_x:
            min_x = x
        if x > max_x:
            max_x = x

    # length of horizontal cable
    cable_length += (max_x - min_x)
    
    # calculate y median to minimize the length of vertical cable(s)
    y_coordinates.sort()
    y_median = y_coordinates[nb_buildings // 2]
    
    # length of vertical cable(s)
    for y in y_coordinates:
        cable_length += math.fabs(y_median - y)
    
    print(int(cable_length))

File: puzzles/go/network-cabling.go
package main

import (
  "fmt"
  "math"
  "sort"
)

func main() {
  var minX = 1073741824.0
  var maxX = 0.0
  var cableSize = 0.0
  var coordinates []float64

  var nbBuildings int
  fmt.Scan(&nbBuildings)
    
  for i := 0; i < nbBuildings; i++ {
    var x, y float64
    fmt.Scan(&x, &y)
    coordinates = append(coordinates, y)
    if x < minX {
      minX = x
    }
    if x > maxX {
      maxX = x
    }
  }

  sort.Float64s(coordinates)
  cableSize = maxX - minX
  
  // To minimize the total vertical length of the cables,
  // the y value of the main cable must be the median of all the y values.
  var median = coordinates[nbBuildings / 2]
  for i := 0; i < nbBuildings; i++ {
    cableSize += math.Abs(median - coordinates[i])
  }

  fmt.Println(int(cableSize))
}
